src/ConnectedMode.UnitTests/BoundSolutionGitMonitorTests.cs

```diff
@@ -0,0 +1,121 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Integration.UnitTests;
+using static SonarLint.VisualStudio.ConnectedMode.BoundSolutionGitMonitor;
+
+namespace SonarLint.VisualStudio.ConnectedMode.UnitTests
+{
+    [TestClass]
+    public class BoundSolutionGitMonitorTests
+    {
+        [TestMethod]
+        public void MefCtor_CheckIsExported()
+        {
+            MefTestHelpers.CheckTypeCanBeImported<BoundSolutionGitMonitor, IBoundSolutionGitMonitor>(
+                MefTestHelpers.CreateExport<IGitWorkspaceService>());
+        }
+
+        [TestMethod]
+        public void Initialize_ForwardsLowLevelEvent()
+        {
+            string repoPath = "some path";
+            int counter = 0;
+
+            var gitWorkspaceService = CreateGitWorkSpaceService(repoPath);
+
+            var gitEventsMonitor = new Mock<IGitEvents>();
+
+            GitEventFactory gitEventFactory = (string path) =>
+            {
+                if (path != repoPath)
+                {
+                    throw new Exception();
```

Comment:
It would be better to include a message in the error explaining that this is a test failure and why i.e. unexpected repo name supplied to the factory.

Alternatively, a simple approach would be just to store the supplied path and assert later that it matches the expected path.